`name-defined` is whitespace-dependent

**Bug Report**

In an attrs class with a field that is named the same as an import, the `name-defined` check is whitespace-dependent.

**To Reproduce**

```python
import httpx
from attrs import define, field


@define
class Foo:
    httpx: httpx.AsyncClient = field(
        factory=lambda: httpx.AsyncClient()
    )
```

**Expected Behavior**

This code should type check identically to:

```python
import httpx
from attrs import define, field


@define
class Foo:
    httpx: httpx.AsyncClient = field(factory=lambda: httpx.AsyncClient())  # on one line
```

**Actual Behavior**

```
test.py:7: error: Name "httpx.AsyncClient" is not defined  [name-defined]
```

**Your Environment**

- Mypy version used: 1.17.1
- Python version used: 3.13.6
- attrs version used: 25.3.0

Attrs is a red herring here.

MRE:

```python
# foo.py
class Bar:
    pass
```

```python
# demo.py
import foo

def nop(fn: object) -> foo.Bar:
    return foo.Bar()

class Fail:
    foo: foo.Bar = nop(
        lambda: 0
    )

class Ok:
    foo: foo.Bar = nop(lambda: 0)
```

```
$ mypy demo.py
demo.py:7: error: Name "foo.Bar" is not defined  [name-defined]
Found 1 error in 1 file (checked 1 source file)
```

Note that the lambda is important (so an extra deferral pass has to do something with this), replacing it with just `0` eliminates the diagnostic. Also note that this only works when the file has `from __future__ import annotations` (otherwise this will be a runtime error, if my in-brain interpreter is not yet dead today), but mypy never checks for that. Whitespace-dependent results are certainly incorrect. 

Maybe this should be P1, but I still don't feel confident assigning priority labels here.

--- a/mypy/semanal.py
+++ b/mypy/semanal.py
@@ -5332,6 +5332,7 @@ class SemanticAnalyzer(
         s.expr.accept(self)
 
     def visit_return_stmt(self, s: ReturnStmt) -> None:
+        old = self.statement
         self.statement = s
         if not self.is_func_scope():
             self.fail('"return" outside function', s)
@@ -5339,6 +5340,7 @@ class SemanticAnalyzer(
             self.fail('"return" not allowed in except* block', s, serious=True)
         if s.expr:
             s.expr.accept(self)
+        self.statement = old
 
     def visit_raise_stmt(self, s: RaiseStmt) -> None:
         self.statement = s

--- a/test-data/unit/check-classes.test
+++ b/test-data/unit/check-classes.test
@@ -9258,3 +9258,18 @@ main:5: note:     def __eq__(self, other: object) -> bool:
 main:5: note:         if not isinstance(other, C):
 main:5: note:             return NotImplemented
 main:5: note:         return <logic to compare two C instances>
+
+[case testLambdaInAttributeCallValue]
+# https://github.com/python/mypy/issues/19632
+import foo
+
+def nop(fn: object) -> foo.Bar:
+    return foo.Bar()
+
+class Bar:
+    foo: foo.Bar = nop(
+        lambda: 0
+    )
+[file foo.py]
+class Bar:
+    ...

